src/redecomp/RedecompMesh.cpp

```diff
@@ -125,13 +127,18 @@ RedecompMesh::RedecompMesh(
 
 EntityIndexByRank RedecompMesh::BuildP2RElementList(
   const Partitioner& partitioner,
-  int n_parts
+  int n_parts,
+  double ghost_length
 ) const
 {
+  if (ghost_length < 0.0)
```

Comment:
Out of curiosity, what would happen if ghost length was 0.0? Or if ghost length was some small positive number that wasn't the 'length' of adjacent ghost elements? Maybe we guard for this possibility?

Reply:
This should be fine -- there would be no ghost elements if the ghost length is 0.0.  Created a few new constructors so the ghost length based on the max element size isn't based on the value of the ghost length passed in.